lib/propmangle.js

```diff
@@ -150,6 +150,11 @@ function mangle_properties(ast, options) {
             if (!(keep_quoted && node.quote))
                 node.key = mangle(node.key);
         }
+        else if (node instanceof AST_ConciseMethod && node.name && node.key.name) {
```

Comment:
I've updated the patch for #2262, and I think this should cover it:
https://github.com/mishoo/UglifyJS2/pull/2262/files#diff-7d9849d6584d8e1e4c321aca78e0977aR96

PTAL